modules/aruco/src/aruco.cpp

```diff
@@ -568,17 +576,21 @@ static int _getBorderErrors(const Mat &bits, int markerSize, int borderSize) {
  *                           2 if the candidate is a white candidate
  */
 static uint8_t _identifyOneCandidate(const Ptr<Dictionary>& dictionary, InputArray _image,
-                                  vector<Point2f>& _corners, int& idx,
-                                  const Ptr<DetectorParameters>& params, int& rotation)
+                                  const vector<Point2f>& _corners, int& idx,
+                                  const Ptr<DetectorParameters>& params, int& rotation,
+                                  const float scale = 1.f)
 {
-    CV_Assert(_corners.size() == 4);
-    CV_Assert(_image.getMat().total() != 0);
-    CV_Assert(params->markerBorderBits > 0);
```

Comment:
these conditions were checked already (in detectMarkers)

Reply:
It makes sense to keep them as `CV_DbgAssert()` anyway.

There are several static code analyzers which don't care about callers code but needs some checks.

Reply:
added `CV_DbgAssert`()